test/pylib/scylla_cluster.py

```diff
@@ -698,6 +721,33 @@ def update_config(self, server_id: ServerNum, key: str, value: object) -> Action
         self.servers[server_id].update_config(key, value)
         return ScyllaCluster.ActionReturn(success=True)
 
+    async def change_ip(self, server_id: ServerNum) -> ActionReturn:
+        """Lease a new IP address and update conf/scylla.yaml with it. The
+        original IP is released at the end of the test to avoid recycle
+        within the existing cluster (scylla currently doesn't support
+        it)."""
+        if server_id not in self.servers:
+            return ScyllaCluster.ActionReturn(success=False, msg=f"Server {server_id} unknown")
+        server = self.servers[server_id]
+        if server.is_running:
+            msg = f"Server {server_id} is running: stop fist it and then change its ip"
+            return ScyllaCluster.ActionReturn(success=False, msg=msg)
+        self.is_dirty = True
+        ip_addr = IPAddress(await self.host_registry.lease_host())
+        self.leased_ips.add(ip_addr)
+        logging.info("Cluster %s changed server %s IP from %s to %s", self.name,
+                     server_id, server.ip_addr, ip_addr)
+        old_ip_addr = server.ip_addr
```

Comment:
It might be useful to keep `old_ip_addr` somewhere. But this can be added in a later patch.

Reply:
Although I agree it might be useful, for complete tracking you'd need a list. And even a list is trivial to add (I did that for the sake of experiment), but then it sits in Server class without much use. I think it should be added when we need to. 

I also considered whether that would be useful for the sake of tracking that no IP address leaks. We already have this tracking on the artifact level, so if there is a leaked artifact, we'll see it at shutdown.